### ISSUE TITLE
New Widget: Overall Page Metrics

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#4122]
## Feature Description

<img width="881" alt="Screenshot 2021-09-24 at 19 48 34" src="https://user-images.githubusercontent.com/90871/134725438-9f483b4d-cc3a-4f52-839a-04c8212e9802.png">

Based on the existing **Search Funnel Widgets** (see: `dashboardSearchFunnel` Widget Area) create a new widget with 4 `<Sparkline />` components with the following data from Analytics:

  * Pageviews
  * Unique Pageviews
  * Bounce Rate
  * Session Duration

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Create a new Overall Page Metrics widget in `assets/js/modules/analytics/components/dashboard/OverallPageMetricsWidget`.
* Based on the design of the `dashboardSearchFunnel` Widget Area, show four Sparkline graphs with the following Analytics metrics:
  * Pageviews
  * Unique Pageviews
  * Bounce Rate
  * Session Duration
* If Analytics isn’t connected, do not show this widget at all (eg. the component should `return null;` or should use the `whenActive` HOC)
* Note that the old Search Funnel view was actually four separate widgets (https://github.com/google/site-kit-wp/blob/d77d4fc8d20ff6d5fa5e266c9be495c3bc3d8ae3/assets/js/modules/search-console/index.js#L60-L79 and https://github.com/google/site-kit-wp/blob/d77d4fc8d20ff6d5fa5e266c9be495c3bc3d8ae3/assets/js/modules/analytics/index.js#L84-L104). But since all data is from one source, these don’t need to be four separate widgets but instead can be combined into one full-width widget that uses a grid internally to responsively size itself for mobile/tablet viewports.
* The content/report should be site-wide for the Main Dashboard, and entity-specific for the Entity Dashboard.
* This new widget should be registered when the `unifiedDashboard` feature flag is enabled, specifically added to the `mainDashboardContent` and `entityDashboardContent` widget areas as full-width widgets. It should appear as the third widget in each of these widget areas. See order in #4078.

## Implementation Brief

* Create a new `DashboardOverallPageMetricsWidget` widget in `analytics/components/dashboard/DashboardOverallPageMetricsWidget`—it should be functionally/visually similar to the `dashboardSearchFunnel` Widget Area—see the ACs. Use the `whenActive` HOC to make sure that the widget is not rendered if the Analytics module is not active or not connected.
  * Grab `isGatheringData` status using the `select( MODULES_ANALYTICS ).isGatheringData()` selector.
  * Request a report with 4 metrics: `ga:pageviews`, `ga:uniquePageviews`, `ga:avgSessionDuration`, and `ga:bounceRate`.
    * Add the current entity URL to the report arguments if it is not empty.
  * Return the loading state if the report hasn't been resolved yet or if the gathering data equals `undefined`.
  * Return the zero state if the `isGatheringData` variable euqals `true` or the `isZeroReport` function call returns `true`
  * Return the expected widget view with report data otherwise. Use widgets mentioned in AC as an example how to create the requested view.
* Register the new widget with the following parameters, when the `unifiedDashboard` feature flag is enabled:
  * `slug` should be `analyticsOverallPageMetrics`
  * `width` should be `FULL`
  * `priority` should be `3`
  * `widgetAreas` array should contain `AREA_MAIN_DASHBOARD_TRAFFIC_PRIMARY` and `AREA_ENTITY_DASHBOARD_TRAFFIC_PRIMARY` area slugs
* Add the new widget to the storybook.

### Test Coverage

* NA

## QA Brief

Note: Please read [this comment](https://github.com/google/site-kit-wp/issues/4122#issuecomment-954041311) which clarifies a couple of points about the AC.

For reference here's a screen grab of the new widget:

![image](https://user-images.githubusercontent.com/18395600/139422277-2e9e3566-1229-4a17-9385-8c9b5efeb261.png)

* Install the [Tester Plugin](https://sitekit.withgoogle.com/service/download/google-site-kit-tester.zip) and ensure the Unified Dashboard feature is enabled:
![image](https://user-images.githubusercontent.com/18395600/139421032-d36098a0-4df2-4eb5-9065-e761a563ead7.png)
* Navigate to the Dashboard, the new Overall Page Metrics widget should be displayed as per the AC.
* Navigate to an Entity Dashboard and verify the Overall Page Metrics widget is displayed as per the AC.

## Changelog entry

* Add the new Overall Page Metrics widget to unified dashboards.

**Comment by felixarntz** [comment]
@tofumatt The ACs here are somehow a duplicate off #4123 - _that_ one is actually about the search funnel. Sticking to the (correct) issue description here, this issue is not about the search funnel, it is about "overall page metrics", which are all Analytics metrics. The only thing this has to do with the search funnel is that it should visually look like today's search funnel widget, but semantically it is unrelated.

Can you rewrite the ACs here to focus on the "overall page metrics" widget? Also the title should probably be changed as it has nothing really to do with a search funnel. The widget JS file here can live in the Analytics module, since it only pulls stats from Analytics.

**Comment by tofumatt** [comment]
@felixarntz Sounds good—as mentioned I think the mix-up between the Design Doc/PRD and the Figma designs sometimes meant I got these shuffled around in my head—though here I thought I updated this, and maybe I just didn't hit "submit" on my changes or something 😓 

At any rate, I've updated the ACs to include the right info for this widget, and include registering it once created. If these ACs look good feel free to move it over to IB 👍🏻

**Comment by felixarntz** [comment]
@tofumatt No worries, looks good now! Only a few comments where I think it could be clarified:

> Create a new Overall Page Metrics widget in `assets/js/ANALYTICS`.

Is that a placeholder? Maybe we should specify something like `assets/js/modules/analytics/components/dashboard/OverallPageMetricsWidget`.

> If Analytics isn’t connected, show a "connect Analytics CTA"

That's not something we've defined before on how it should be, but I think we should rephrase this to not show the entire widget at all if Analytics is not active/connected, for two reasons:
* It's no longer only partially Analytics like in the Search Funnel, now the entire 4 metrics require Analytics anyway.
* We already would display a CTA to activate Analytics when this widget is shown in another widget above, so hiding this widget here instead results in a less cluttered UI.

Let me know if that sounds good to you, and if so could you update the ACs? Other than that lgtm!

**Comment by tofumatt** [comment]
Ah, shoot, that was indeed a placeholder or a typo! Fixed the file issue.

Regarding not showing the widget at all: that seems fine to me. I thought the CTA might be useful but true: it's not in the original doc and your reasoning here makes sense. I've amended the ACs so I think this is good; moving to IB 👍🏻

**Comment by techanvil** [comment]
Hi @cole10up, you might have already spotted this, but I have noticed that I have missed out the unit character(s) for Bounce Rate and Session Duration when implementing the Overall Page Metrics widget, as is apparent when comparing it to a similar widget.

Overall Page Metrics widget:
![image](https://user-images.githubusercontent.com/18395600/140293121-63f1f19d-8480-483c-8514-e8a401233df2.png)

Vs Analytics Overview widget:
![image](https://user-images.githubusercontent.com/18395600/140293227-ad192e7a-0ad6-4383-9d91-bbcd27e1d543.png)

**Comment by techanvil** [comment]
@cole10up re. the above, I have created a separate ticket for the unit characters fix: https://github.com/google/site-kit-wp/issues/4312

**Comment by cole10up** [comment]
### **QA ❌**

Thanks for logging the character fix @techanvil 

I was able to confirm statistics on the new widget:
![image](https://user-images.githubusercontent.com/57115907/140663562-97b527cc-0f9a-4339-9419-306b1c837beb.png)

Verified the Analytics link.

I noticed with the unified dashboard feature flag on - the analytics and search console pages are showing blank:
![image](https://user-images.githubusercontent.com/57115907/140663583-8503d83f-62b8-42db-b439-66eb1a167f71.png)

![image](https://user-images.githubusercontent.com/57115907/140663592-0207004e-a881-4da8-96eb-9ff6eb85be43.png)

Flipping the feature flag off shows these pages with proper content.

Sending back to @techanvil to check.

**Comment by aaemnnosttv** [comment]
> I noticed with the unified dashboard feature flag on - the analytics and search console pages are showing blank:

@cole10up this is expected as these pages are going away with the unified dashboard. See https://github.com/google/site-kit-wp/issues/4134

Sending back to @wpdarren in case there's anything left to review here 👍

**Comment by wpdarren** [comment]
I will give this a run through as a sanity check and then approve. 😄

**Comment by wpdarren** [comment]
### QA Update: ✅ 

I've sent this to approved. Cole previously confirmed that the only issue was blank module pages. 

@felixarntz @tofumatt one small observation: I did notice in the AC, widgets were ordered: Session Duration
then Bounce Rate, but the widgets are displayed the other way around on the screenshots above.  Reading through the comments it looks like a change was made but AC not updated.

**Comment by felixarntz** [comment]
@wpdarren You're right - I've now updated this in the ACs as well, for any future reference.
